src/dynimage.rs

```diff
@@ -78,6 +78,39 @@ pub enum DynamicImage {
     ImageRgba32F(Rgba32FImage),
 }
 
+impl Clone for DynamicImage {
+    fn clone(&self) -> Self {
+        match self {
+            Self::ImageLuma8(p) => Self::ImageLuma8(p.clone()),
+            Self::ImageLumaA8(p) => Self::ImageLumaA8(p.clone()),
+            Self::ImageRgb8(p) => Self::ImageRgb8(p.clone()),
+            Self::ImageRgba8(p) => Self::ImageRgba8(p.clone()),
+            Self::ImageLuma16(p) => Self::ImageLuma16(p.clone()),
+            Self::ImageLumaA16(p) => Self::ImageLumaA16(p.clone()),
+            Self::ImageRgb16(p) => Self::ImageRgb16(p.clone()),
+            Self::ImageRgba16(p) => Self::ImageRgba16(p.clone()),
+            Self::ImageRgb32F(p) => Self::ImageRgb32F(p.clone()),
+            Self::ImageRgba32F(p) => Self::ImageRgba32F(p.clone()),
+        }
+    }
+
+    fn clone_from(&mut self, source: &Self) {
+        match (self, source) {
+            (Self::ImageLuma8(p1), Self::ImageLuma8(p2)) => p1.clone_from(p2),
+            (Self::ImageLumaA8(p1), Self::ImageLumaA8(p2)) => p1.clone_from(p2),
+            (Self::ImageRgb8(p1), Self::ImageRgb8(p2)) => p1.clone_from(p2),
+            (Self::ImageRgba8(p1), Self::ImageRgba8(p2)) => p1.clone_from(p2),
+            (Self::ImageLuma16(p1), Self::ImageLuma16(p2)) => p1.clone_from(p2),
+            (Self::ImageLumaA16(p1), Self::ImageLumaA16(p2)) => p1.clone_from(p2),
+            (Self::ImageRgb16(p1), Self::ImageRgb16(p2)) => p1.clone_from(p2),
+            (Self::ImageRgba16(p1), Self::ImageRgba16(p2)) => p1.clone_from(p2),
+            (Self::ImageRgb32F(p1), Self::ImageRgb32F(p2)) => p1.clone_from(p2),
+            (Self::ImageRgba32F(p1), Self::ImageRgba32F(p2)) => p1.clone_from(p2),
+            (this, source) => *this = source.clone(),
```

Comment:
This match arm should be avoided. The enum is marked non-exhaustive, but within this crate we can still match exhaustively on it

Reply:
I am not sure of what you want here. Something like this ?

```rust
match (self, source) {
    (Self::ImageRgb8(p1), Self::ImageRgb8(p2)) => p1.clone_from(p2),
    (Self::ImageRgba8(p1), Self::ImageRgba8(p2)) => p1.clone_from(p2),
    // other formats...

    (this, Self::ImageRgb8(p)) => *this = Self::ImageRgb8(p.clone()),
    (this, Self::ImageRgba8(p)) => this = Self::ImageRgba8(p.clone()),
    // other formats...
}
```

Reply:
Sorry, I didn't read closely. It's fine as is